src/pdn/src/PdnGen.tcl

```diff
@@ -3230,36 +3273,6 @@ proc generate_via_stacks {l1 l2 tag connection} {
     set width [expr $xMax - $xMin]
     set height [expr $yMax - $yMin]
 
-    # Ensure that the intersections are not partial
-    if {![regexp {(.*)_PIN_(hor|ver)} $l1]} {
-      if {[get_dir $layer1] == "hor"} {
-        if {$height < [get_grid_wire_width $layer1]} {
-          # If the intersection doesnt cover the whole width of the bottom level wire, then ignore
-          utl::warn "PDN" 40 "No via added at ([expr 1.0 * $xMin / $def_units] [expr 1.0 * $yMin / $def_units] [expr 1.0 * $xMax / $def_units] [expr 1.0 * $yMax / $def_units]) because the full height of $layer1 ([expr 1.0 * [get_grid_wire_width $layer1] / $def_units]) is not covered by the overlap."
-          continue
-        }
-      } else {
-        if {$width < [get_grid_wire_width $layer1]} {
-          # If the intersection doesnt cover the whole width of the bottom level wire, then ignore
-          utl::warn "PDN" 41 "No via added at ([expr 1.0 * $xMin / $def_units] [expr 1.0 * $yMin / $def_units] [expr 1.0 * $xMax / $def_units] [expr 1.0 * $yMax / $def_units]) because the full width of $layer1 ([expr 1.0 * [get_grid_wire_width $layer1] / $def_units]) is not covered by the overlap."
-          continue
-        }
-      }
-    }
-    if {[get_dir $layer2] == "hor"} {
-      if {$height < [get_grid_wire_width $layer2]} {
-        # If the intersection doesnt cover the whole width of the top level wire, then ignore
-        utl::warn "PDN" 42 "No via added at ([expr 1.0 * $xMin / $def_units] [expr 1.0 * $yMin / $def_units] [expr 1.0 * $xMax / $def_units] [expr 1.0 * $yMax / $def_units]) because the full height of $layer2 ([expr 1.0 * [get_grid_wire_width $layer2] / $def_units]) is not covered by the overlap."
-        continue
-      }
-    } else {
-      if {$width < [get_grid_wire_width $layer2]} {
-        # If the intersection doesnt cover the whole width of the top level wire, then ignore
-        utl::warn "PDN" 43 "No via added at ([expr 1.0 * $xMin / $def_units] [expr 1.0 * $yMin / $def_units] [expr 1.0 * $xMax / $def_units] [expr 1.0 * $yMax / $def_units]) because the full width of $layer2 ([expr 1.0 * [get_grid_wire_width $layer2] / $def_units]) is not covered by the overlap."
-        continue
-      }
-    }
-
```

Comment:
This code specifically checks that in order to connect two layers together, that there is a full overlap of the lines. A parial overlap can result in problems with via sizing, and are likely the result of a poor floorplan, or interaction of macros cutting channels into the design.
I believe this code needs to be retained.